org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchBugs9Tests.java

```diff
@@ -4800,6 +4800,32 @@ public void testMethodReferenceForTypeFromJREModuleBugGh740() throws Exception {
 		deleteProject(projectName);
 	}
 }
+public void testGH902_whenTypeReferenceIsUnknown_expectToBeFound() throws CoreException {
+	try {
+		IJavaProject project = createJava9Project("JavaSearchBugs9");
+		project.open(null);
+		createFile("JavaSearchBugs9/src/GH902.java",
+		"""
+			public class GH902 {
```

Comment:
Please add a test case for nested types, too:

```
package dummy;

import java.util.AbstractMap;

public abstract class GH902 extends AbstractMap<String, String> {
	public static void foo() {
		Entry.comparingByKey();
	}
}
```
If a search is performed for `Entry` with the latest Eclipse release, this usage isn't revealed. If I understand the comment below about java.lang types correctly, this fix might not be sufficient. In any case, an additional test might be useful.

Reply:
Also of interest:
```
public class GH902 {
	public static void foo() {
		java.util.AbstractMap.Entry.comparingByKey();
	}
}
```

Reply:
```
package dummy;

import static java.util.AbstractMap.*;

public class GH902 {
	public static void foo(Object o) {
		Entry.comparingByKey();
	}
}
```

Reply:
Also relevant when a constructor instead of a static method was used. Reference to AbstractMap.SimpleEntry are not found:
```
package dummy;

import java.util.HashMap;

public class GH902 {
	public static void foo(Object o) {
		new HashMap.SimpleEntry<>(null, null);
	}
}
```

Reply:
I added few tests for nested scenarios and remove the java.lang comment and relevant changes. Seems it was a workspace fluke the simple name as qualified. probably unresolved project. But i have added tests for qualified scenarios as well.

Reply:
Did you see the test failures?

Reply:
Yes, thats was my mistake, forgot to change the search type and assertions. Now i have done the changes, But i couldn't run them locally due to my workspace is broken for some reason. If it fails again, this has to wait until i setup a new workspace for jdt.